packages/app-shell-shared/src/hooks/useFilteredConfig.ts

```diff
@@ -0,0 +1,445 @@
+
+    return () => {
+      cancelled = true;
+    };
+  }, [bundleList]);
+
+  return { hooks, isLoading };
+};
+
+/**
+ * Filter config based on condition results using element keys
+ */
+export const filterConfigWithResults = (
+  config: HvAppShellConfig,
+  metadata: ConditionMetadata[],
+  results: Map<number, boolean>,
+): HvAppShellConfig => {
+  // Group metadata by element key for efficient AND logic
+  const metadataByElement = new Map<string | number, ConditionMetadata[]>();
+
+  for (const meta of metadata) {
+    const group = metadataByElement.get(meta.elementKey) ?? [];
+    group.push(meta);
+    metadataByElement.set(meta.elementKey, group);
+  }
+
+  // Build element results map with AND logic: all conditions must be true
+  const elementResults = new Map<string | number, boolean>();
+
+  for (const [elementKey, metas] of metadataByElement) {
+    const allTrue = metas.every((meta) => results.get(meta.hookIndex) ?? false);
+    elementResults.set(elementKey, allTrue);
+  }
+
+  const filterViews = (
+    views: HvAppShellViewsConfig[],
+  ): HvAppShellViewsConfig[] => {
+    const result: HvAppShellViewsConfig[] = [];
+
+    for (const view of views) {
+      if (view.$key === undefined) {
+        result.push(view);
+        continue;
+      }
+
+      const shouldInclude = elementResults.get(view.$key) ?? true;
+
+      if (!shouldInclude) {
+        continue;
+      }
+
+      if (view.views) {
+        result.push({
+          ...view,
+          views: filterViews(view.views),
+        });
+      } else {
+        result.push(view);
+      }
+    }
+
+    return result;
+  };
+
+  const filterMenus = (
+    menus: HvAppShellMenuConfig[],
+  ): HvAppShellMenuConfig[] => {
+    const result: HvAppShellMenuConfig[] = [];
+
+    for (const menu of menus) {
+      if (menu.$key === undefined) {
+        result.push(menu);
+        continue;
+      }
+
+      const shouldInclude = elementResults.get(menu.$key) ?? true;
+
+      if (!shouldInclude) {
+        continue;
+      }
+
+      if (menu.submenus) {
+        const filteredSubmenus = filterMenus(menu.submenus);
+        // Keep parent only if it has visible children OR has its own target
+        if (filteredSubmenus.length > 0 || menu.target) {
+          result.push({
+            ...menu,
+            submenus:
+              filteredSubmenus.length > 0 ? filteredSubmenus : undefined,
+          });
+        }
+      } else {
+        result.push(menu);
+      }
+    }
+
+    return result;
+  };
+
+  const filterActions = (
+    actions: HvAppShellHeaderAction[],
+  ): HvAppShellHeaderAction[] => {
+    return actions.filter((action) => {
+      if (action.$key === undefined) return true;
+      return elementResults.get(action.$key) ?? true;
+    });
+  };
+
+  const filterProviders = (
+    providers: HvAppShellProvidersConfig[],
+  ): HvAppShellProvidersConfig[] => {
+    return providers.filter((provider) => {
+      if (provider.$key === undefined) return true;
```

Comment:
Not sure the use case, but for consistency, .. all providers should have `$key`...